pom.xml

```diff
@@ -827,8 +827,7 @@
       <dependency>
         <groupId>com.google.protobuf</groupId>
         <artifactId>protobuf-java</artifactId>
-        <version>${protobuf.hadoopDependency.version}</version>
-        <scope>${hadoop.deps.scope}</scope>
+        <version>${protobuf.version}</version>
```

Comment:
I have done maven test with `-Phadoop-2` and `-Phadoop-3`, and all tests passed.

If feasible, we don't need to explicitly add `protobuf-java` dependency and declare `protobuf.version` in each module that requires `protobuf-java` 3.x(such as core, sql, connect and protobuf), but this also changes `protobuf-java` version in `dev/deps/spark-deps-hadoop-x-hive-2.3` from 2.5.0 to 3.x

Reply:
Of course, it may be safer to remove `-Phadoop-2` completely before doing this work